### PR TITLE
use MIT license

### LINKED ISSUE #37797: Switch to MIT license
I suggest that Roslyn's license should be changed to a MIT license. This would [alleviate legal problems for developers](https://github.com/dotnet/fsharp/issues/1727) and be consistent with the rest of .NET Core.

### PR DESCRIPTION
**Pull request by jmarolf**
fixes #37797

### CODE REVIEW

**Comment by tannergooding** [comment]
@jmarolf, what about the license header in every cs file?

CC. @richlander

**Comment by sharwell** [comment]
The license headers in C# and VB source files will be updated separately, using the new file header analyzer that @JoeRobich is working on (#33012).

**Comment by 333fred** [comment]
> The license headers in C# and VB source files will be updated separately, using the new file header analyzer that @JoeRobich is working on (#33012).

We should definitely not merge this until our headers are consistent with the license they're referring to in the root of the repo.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "git.ignoreLimitWarning": true
+}
```
Comment (JoeRobich) [inline .vscode/settings.json]: Was adding this file intentional?

Comment (jmarolf) [reply .vscode/settings.json]: nope let me fix that!

**Comment by RikkiGibson** [comment]
So when we flow this into feature branches, where new files added now have an outdated license header, the build will fail, right?

**Comment by sharwell** [comment]
@RikkiGibson No, we do not have file header validation in our build right now (never have actually). It's part of the reason several files in the repository are missing headers.

**Comment by jmarolf** [comment]
Yeah, this was best effort on my part. Once we have enforcement presumably there will be a subsequent PR that will add headers to files that don't have them today. The only way this could break you is if you were modifying the license expressions in the targets files in a feature branch.

**Pull request merged**